jsx/Form.js

```diff
@@ -456,6 +456,71 @@ var DateElement = React.createClass({
   }
 });
 
+/**
+ * Time Component
+ * React wrapper for a <input type="time"> element.
+ */
+var TimeElement = React.createClass({
+
+  propTypes: {
+    name: React.PropTypes.string.isRequired,
+    label: React.PropTypes.string,
+    value: React.PropTypes.string,
+    id: React.PropTypes.string,
+    disabled: React.PropTypes.bool,
+    required: React.PropTypes.bool,
+    onUserInput: React.PropTypes.func
+  },
+
+  getDefaultProps: function() {
+    return {
+      name: '',
+      label: '',
+      value: '',
+      id: null,
+      disabled: false,
+      required: false,
+      onUserInput: function() {
+        console.warn('onUserInput() callback is not set');
+      }
+    };
+  },
+  handleChange: function(e) {
+    this.props.onUserInput(this.props.name, e.target.value);
+  },
+  render: function() {
+    var disabled = this.props.disabled ? 'disabled' : null;
+    var required = this.props.required ? 'required' : null;
+    var requiredHTML = null;
+
+    // Add required asterix
+    if (required) {
+      requiredHTML = <span className="text-danger">*</span>;
+    }
+
+    return (
+      <div className="row form-group">
+        <label className="col-sm-3 control-label" htmlFor={this.props.label}>
+          {this.props.label}
+          {requiredHTML}
+        </label>
+        <div className="col-sm-9">
+          <input
+            type="time"
```

Comment:
How does this work across browsers? IIRC the date/time HTML5 elements are poorly supported by firefox, which falls back to displaying a text input box.. so you might want to add some validation.

Also, what format does the time get submitted to the backend when someone submits using this element type?

Reply:
According to w3, the 'time' input is supported by internet explorer, firefox, chrome, safari and opera. 

![image](https://user-images.githubusercontent.com/7550451/35690409-ca8aab52-0743-11e8-96bd-23dcbdab9b75.png)

Although developer.mozilla claims that safari and internet explorer are not supported

![image](https://user-images.githubusercontent.com/7550451/35690765-b4adfc5c-0744-11e8-9e61-d031806939d7.png)

Also from the mozilla website:
"When a form including a time input is submitted, the value is encoded before being included in the form's data. The form's data entry for a time input will always be in the form "name=hh%3Amm", or "name=hh%3Amm%3ass" if seconds are included."

Reply:
That's I don't think it's safe to assume that all Firefox users have already upgraded to Firefox 57 at this point. (Though it's good to know they finally added support!)

Reply:
In that case, what do you suggest as an alternative for that would function as a time input?

Reply:
@driusan just checking that I should find an alternative way of creating a time input for LORIS react forms. 

I think one of the other options would be that I keep the component as is, but add a fall-back validation that checks that the input is in the proper format, in case the user's version of firefox defaults to a simple text field. Alternatively, by the time any forms that use this element get merged into LORIS, perhaps it will be safe to assume or require that the user be upgraded to Firefox 57.

Reply:
There should be a polyfill for older Firefox versions already in LORIS, we just need to be careful to validate the time in the backend and document what the expected format is.

Reply:
Would you like to see some validation/documentation changes in this PR specifically? Or would that be the responsibility of the given back-end submission code that is implementing this form element?

Reply:
It's the responsibility of the backend code using it. If you verify that we have a polyfill, this PR should be fine. Sorry for the confusion.. there have just been problems before when trying to use HTML5 date/time elements because Firefox was really, really, really slow to implement them.

Reply:
Sounds good! I will very that we have a polyfill and get back to you!

Reply:
i don't think we have a polyfill for time inputs

Reply:
Okay! I'm on it.

Reply:
if the polyfill option doesn't work out for you, i'd recommend taking a look at the "Handling browser support" section of https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time